conf/modules/logger_sd_spi_direct.xml

```diff
@@ -0,0 +1,61 @@
+<!DOCTYPE module SYSTEM "module.dtd">
+
+<module name="logger_sd_spi_direct" dir="loggers">
+  <doc>
+    <description>Direct SPI SD Logger that saves pprzlog messages to SD Card.</description>
+    <configure name="SDLOGGER_DIRECT_SPI" value="SPI1|SPI2|SPI3|SPI4|SPI5|SPI6" description="Port to which the SD Card is connected."/>
+    <configure name="SDLOGGER_DIRECT_SPI_SLAVE" value="SPI_SLAVE1|SPI_SLAVE2|SPI_SLAVE3|SPI_SLAVE4|SPI_SLAVE5|SPI_SLAVE6" description="Port to which the SD Card is connected."/>
+    <configure name="LOGGER_CONTROL_SWITCH" value="RADIO_AUX2"/>
+    <configure name="LOGGER_LED" value="none"/>
+  </doc>
+  <settings>
+    <dl_settings NAME="sdlogger">
+      <dl_settings NAME="SD Logger">
+        <dl_setting module="loggers/sdlogger_spi_direct" var="sdlogger_spi.status" min="0" max="20" step="1"/>
+        <dl_setting module="loggers/sdlogger_spi_direct" var="sdcard1.status" min="0" max="40" step="1"/>
+        <dl_setting module="loggers/sdlogger_spi_direct" var="sdcard1.error_status" min="0" max="15" step="1"/>
+        <dl_setting module="loggers/sdlogger_spi_direct" var="sdlogger_spi.command" min="1" max="42" step="1" />
+      </dl_settings>
+    </dl_settings>
+  </settings>
+  <header>
+    <file name="sdlogger_spi_direct.h"/>
+  </header>
+  <init fun="sdlogger_spi_direct_init()"/>
+  <periodic fun="sdlogger_spi_direct_periodic()" freq="512" start="sdlogger_spi_direct_start()" stop="sdlogger_spi_direct_stop()" autorun="TRUE"/>
+  <datalink message="SETTING" fun="sdlogger_spi_direct_command()"/>
+  <makefile>
+
+    <raw>
+        SDLOGGER_DIRECT_SPI ?= spi2
+        SDLOGGER_DIRECT_SPI_LOWER=$(shell echo $(SDLOGGER_DIRECT_SPI) | tr A-Z a-z)
+        SDLOGGER_DIRECT_SPI_UPPER=$(shell echo $(SDLOGGER_DIRECT_SPI) | tr a-z A-Z)
+
+        SDLOGGER_DIRECT_SPI_SLAVE ?= spi_slave2
+        SDLOGGER_DIRECT_SPI_SLAVE_LOWER=$(shell echo $(SDLOGGER_DIRECT_SPI_SLAVE) | tr A-Z a-z)
+        SDLOGGER_DIRECT_SPI_SLAVE_UPPER=$(shell echo $(SDLOGGER_DIRECT_SPI_SLAVE) | tr a-z A-Z)
+
+        LOGGER_CONTROL_SWITCH ?= RADIO_AUX2
+
+        LOGGER_LED ?= none
+        ifneq ($(LOGGER_LED),none)
+        ap.CFLAGS += -DLOGGER_LED=$(LOGGER_LED)
+        endif
+
```

Comment:
Please add `include $(CFG_SHARED)/spi_master.makefile` and remove `SPI_MASTER` further down.
That way it will also add the needed sources for spi if you don't already use another subsystem that includes SPI.